x-pack/plugins/ml/server/models/results_service/anomaly_charts.ts

```diff
@@ -0,0 +1,1946 @@
+        const accessor = Object.keys(aggFields)[0];
+        const tempAggs = { ...(aggFields[accessor].aggs ?? aggFields[accessor].aggregations) };
+        const foundValue = findAggField(tempAggs, metricFieldName);
+
+        if (foundValue !== undefined) {
+          tempAggs.metric = foundValue;
+          delete tempAggs[metricFieldName];
+        }
+        esSearchRequest.aggs!.byTime.aggs = tempAggs;
+      } else {
+        esSearchRequest.aggs!.byTime.aggs.metric = metricAgg;
+      }
+    } else {
+      // if metricFieldName is not defined, it's probably a variation of the non zero count function
+      // refer to buildConfigFromDetector
+      if (summaryCountFieldName !== undefined && metricFunction === ES_AGGREGATION.CARDINALITY) {
+        // if so, check if summaryCountFieldName is an aggregation field
+        if (typeof aggFields === 'object' && Object.keys(aggFields).length > 0) {
+          // first item under aggregations can be any name, not necessarily 'buckets'
+          const accessor = Object.keys(aggFields)[0];
+          const tempAggs = { ...(aggFields[accessor].aggs ?? aggFields[accessor].aggregations) };
+          const foundCardinalityField = findAggField(tempAggs, summaryCountFieldName);
+          if (foundCardinalityField !== undefined) {
+            tempAggs.metric = foundCardinalityField;
+          }
+          esSearchRequest.aggs!.byTime.aggs = tempAggs;
+        }
+      }
+    }
+
+    const resp = await client?.asCurrentUser.search(esSearchRequest);
+
+    const obj: MetricData = { success: true, results: {} };
+    // @ts-ignore
+    const dataByTime = resp?.aggregations?.byTime?.buckets ?? [];
+    dataByTime.forEach((dataForTime: any) => {
+      if (metricFunction === 'count') {
+        obj.results[dataForTime.key] = dataForTime.doc_count;
+      } else {
+        const value = dataForTime?.metric?.value;
+        const values = dataForTime?.metric?.values;
+        if (dataForTime.doc_count === 0) {
+          // @ts-ignore
+          obj.results[dataForTime.key] = null;
+        } else if (value !== undefined) {
+          obj.results[dataForTime.key] = value;
+        } else if (values !== undefined) {
+          // Percentiles agg currently returns NaN rather than null when none of the docs in the
+          // bucket contain the field used in the aggregation
+          // (see elasticsearch issue https://github.com/elastic/elasticsearch/issues/29066).
+          // Store as null, so values can be handled in the same manner downstream as other aggs
+          // (min, mean, max) which return null.
+          const medianValues = values[ML_MEDIAN_PERCENTS];
+          obj.results[dataForTime.key] = !isNaN(medianValues) ? medianValues : null;
+        } else {
+          // @ts-ignore
+          obj.results[dataForTime.key] = null;
+        }
+      }
+    });
+
+    return obj;
+  }
+
+  /**
+   * TODO Make an API endpoint (also used by the SMV).
```

Comment:
Can you create an issue to list out the extra endpoints we should be adding - we should do this in a follow-up PR.

Reply:
Created https://github.com/elastic/kibana/issues/128501